global-script.lua

```diff
@@ -2757,12 +2799,30 @@ function upCastPosSizRot(oPos,size,rot,dist,multi,tags)
 end
 ---- Block Square Section
 function setupPlayerArea(params)
-    --Sets position/color for the button, spawns it
-    params.obj.createButton({
+    -- Figure out what color we're supposed to be, or if playerswapping is even allowed.
+    local obj = params.obj
+    obj.createButton({
         label="Energy Cost: 0", click_function="nullFunc",
         position={0,2.24,-11.2}, rotation={0,180,0}, height=0, width=0,
         font_color={1,1,1}, font_size=500
     })
+
+    local color
+    for k, v in pairs(playerBlocks) do
+        if v.guid == obj.guid then
+            color = k
+            break
+        end
+    end
+    obj.setVar("playerColor", color)  -- May be nil
+    if color then
+        obj.createButton({
+            label="Move to here", click_function="onSwapButtonClicked", function_owner=Global,
+            position={0,2.24,-7.2}, rotation={0,180,0}, height=800, width=4000,
+            font_color={0,0,0}, font_size=500, -- hover_color=color,
```

Comment:
can the comment be removed from this line?

Reply:
Already done in my source, which will be in the next push.